lib/liquid/standardfilters.rb

```diff
@@ -377,6 +379,43 @@ def to_date(obj)
       nil
     end
 
+    def nested_respond_to?(input, property)
+      return true if property.empty?
+      p = property.split('.').first
+      q = property[/#{p}.?([\w.]*)/, 1]
+      if input.respond_to?(:[]) && accepts?(input, p)
+        return nested_respond_to?(input[p], q)
+      elsif input.respond_to?(p)
+        return nested_respond_to?(input.send(p), q)
+      end
+      false
+    end
+
+    def nested_send(input, property)
+      return input if property.empty?
+      p = property.split('.').first
+      q = property[/#{p}.?([\w.]*)/, 1]
+      if input.respond_to?(:[]) && accepts?(input, p)
+        return nested_send(input[p], q)
+      elsif input.respond_to?(p)
+        return nested_send(input.send(p), q)
+      end
+      nil
+    end
+
+    def accepts?(input, index)
+      !input[index].nil?
+      rescue TypeError
```

Comment:
I added a number of helper methods that should be called by sort when the property being passed  is nested. I put them with the private methods. Is this the right location for that code?

I would also like to add tests for these helper methods. What is the right place to put these tests? They cannot be called from standard_filter_test.rb through `@filters`